src/middlewared/middlewared/alert/source/snapshot_count.py

```diff
@@ -33,10 +31,10 @@ async def check(self):
         max_total = await self.middleware.call("pool.snapshottask.max_total_count")
 
         total = 0
-        datasets = defaultdict(lambda: 0)
-        for snapshot in await self.middleware.call("zfs.snapshot.query", [], {"select": ["name"]}):
-            total += 1
-            datasets[snapshot["name"].split("@")[0]] += 1
+        datasets = await self.middleware.call("zfs.snapshot.query", [], {"extra": {"count_only": True}})
```

Comment:
It feels weird that a query endpoint is giving a dict - even if we want a count, getting an integer makes more sense rather then a dict. List should do the job nicely ?